pages/api/upcoming/index.js

```diff
@@ -0,0 +1,14 @@
+import fetcher from "@/services/fetch";
+
+export default async function handler(request, response) {
+  try {
+    const { page } = request.query;
+    const url = `https://api.themoviedb.org/3/movie/upcoming?api_key=${process.env.API_KEY}&language=en-US&page=${page}`;
+    const movieData = await fetcher(url);
+    response.status(200).json(movieData);
+  } catch (error) {
+    response
+      .status(500)
+      .json({ error: "An error occured while fetching the data" });
+  }
+}
```

Comment:
Also all of your handler functions are repetitive, so can be outsourced and reused! You can add this also to the refactoring list :)